suse_migration_services/units/product_setup.py

```
@@ -66,53 +68,66 @@ def main():
-                )
+                message = ('Found multiple product definitions'
+                           'without element <flavor>')
+                log.error(message)
+                raise DistMigrationProductSetupException(message)
         log.info('Updating Base Product to be suitable for migration')
-        update_baseproduct(baseproduct)
+        update_baseproduct(baseproduct.prod_filename)
     except Exception as issue:
         message = 'Base Product update failed with {0}'.format(issue)
         log.error(message)
         raise DistMigrationProductSetupException(message)
 
 
 def get_baseproduct(products_metadata):
-    baseproduct_filename = []
-    xml = ElementTree()
-    for product_name in os.listdir(products_metadata):
-        product = os.sep.join(
-            [products_metadata, product_name]
-        )
-        xml.parse(product)
-        register_sections = xml.findall('register')
-        for register in register_sections:
-            flavor = register.findall('flavor')
-            if not flavor:
-                baseproduct_filename.append(product)
+    base_product_type = namedtuple(
+        'base_product_type', ['count', 'prod_filename']
+    )
+    prod_filenames = glob.glob(
+        os.path.join(products_metadata, '*.prod')
+    )
+    count = 0
+    name = None
+    try:
+        xml = ElementTree()
+        for prod_filename in prod_filenames:
+            xml.parse(prod_filename)
+            register_sections = xml.findall('register')
+            for register in register_sections:
+                flavor = register.findall('flavor')
+                if not flavor:
+                    count = count + 1
+                    name = prod_filename
 
-    return baseproduct_filename
+        return base_product_type(count, name)
+    except Exception as issue:
+        message = ('Getting base product for product '
+                   'update failed with {0}'.format(issue))
```

Tee error should just tell us that there is a file that is not correct. This in and off itself is not really a problem, we just move on.

message = 'Could not pars {0} as XML file'.format(prod_filename)

Also the try: execpt needs to be inside the for loop so we can follow a best effort approach, i.e. parse as many files as possible ignore any errors.